client/src/app.jsx

```diff
@@ -1,33 +1,85 @@
   import React, { Component } from 'react';
   import axios from 'axios';
 
+  import CalendarContainer from './CalendarContainer'
+
   export default class App extends Component {
     constructor(props) {
       super(props);
       this.state = {
-        listingId: null
+        houseData: null
       };
-      this.houseData = {};
     };
 
     componentDidMount() {
       if(this.props.listingId) {
         axios.get(`/house?listingId=${this.props.listingId}`)
           .then((response) => {
-            this.houseData = response.data[0];
             this.setState({
-              listingId: this.houseData.listingId
+              houseData: response.data[0]
             });
           });
       };
     };
 
+    getDates() {
```

Comment:
Nice job on the complicated logic. If you ever wanted to fully test it you could move it out of the file and import it into the App. This is more of a preference / option if your App starts to get crowded.  Don't worry about it for now, I imagine this was more logic than you planned.